R/information-matrices.R

```diff
@@ -265,6 +331,30 @@ Fisher_info <- function(mod, type = "expected") {
   }
 
   rownames(info) <- colnames(info) <- theta_names
+
+  # info mat for separate variances
+  if (!is.null(mod$call$weights) && inherits(mod$modelStruct$varStruct, "varIdent") && separate_variances) {
+    theta_reparam <- extract_varcomp(mod, separate_variances = TRUE)
+    theta_reparam_names <- vapply(strsplit(names(unlist(theta_reparam)), split = "[.]"),
+                                  function(x) paste(unique(x), collapse = "."), character(1L))
+    r12 <- length(unlist(theta[c(1,2)]))
```

Comment:
In all of the Jacobean calculations, I think we'll need to refer to `theta` components by name instead of by index. Otherwise, this won't work for `gls` objects. See additional unit tests I just pushed.

Reply:
Thanks for catching this bug and for adding the unit tests for `gls` models. This should be fixed in the new commit. Please let me know if I missed anything else.